include/zephyr/net/wifi_mgmt.h

```diff
@@ -384,7 +384,7 @@ enum wifi_conn_status {
  */
 enum wifi_disconn_reason {
 	/** Unspecified reason */
-	WIFI_REASON_DISCONN_UNSPECIFIED = 0,
+	WIFI_REASON_DISCONN_UNSPECIFIED = 1,
```

Comment:
Thanks for the explanation. Now this causes more questions, I mean why do we return disconnected status when trying to do a connect operation. Shouldn't the status contain a value of `wifi_conn_status`?

Reply:
Yes, ideally for connection operation wifi_conn_status should be used. I will check further and update the changes.
Thanks.

Reply:
When we are trying to do connect operation with Wrong key, it will be deauthenticate and return that disconnected status which is indicating a reason for disconnection.

Reply:
That sounds like a bug. How could the connect call in wifi shell know that the status is from `wifi_disconn_reason`. There is already `WIFI_STATUS_CONN_WRONG_PASSWORD` in `wifi_conn_status` that looks like it maps nicely to this case. Is there a reason why we could not use this code instead?

Reply:
Not quite, the sequence is below:

1. connect_request
2. connect_failure (WRONG_KEY)
3. unsolicited disconn : As AP sends diassociation/deauthentication

Because we have a 121 mapping between protocol and Wi-Fi management events, 3 is possible.

Reply:
Ok, after looking what the `struct wifi_status` look like I can see the issue. We try to map all the status codes from different operations to one status. This works ok if we try to do a connect and get returned connect status, but this fails if we get returned disconnect status when connecting. We either need to return connect status for connect operation, or we need to use unique values for all the operations. But imho we cannot mix the codes between different operations, this is just super confusing for the end user.

Reply:
But isn't it (confusion about the error code) what was caused this PR to be created in the first place?

Reply:
Yes, for the `disconnect_result` sending using `UNSPECIFIED_REASON` was causing it to print as `disconnect successful`. Basically like 802.11, we could have separate `status codes` (solicited) and `reason codes` (unsolicited), to properly fix this, but fixing the enum would still help in re-using the same set as both.

Reply:
We could keep this simple and use one enum space for the status values. So something like this could be done:
```
diff --git a/include/zephyr/net/wifi_mgmt.h b/include/zephyr/net/wifi_mgmt.h
index e0930d2f106..e11881ee7ce 100644
--- a/include/zephyr/net/wifi_mgmt.h
+++ b/include/zephyr/net/wifi_mgmt.h
@@ -377,6 +377,10 @@ enum wifi_conn_status {
        WIFI_STATUS_CONN_TIMEOUT,
        /** Connection failed - AP not found */
        WIFI_STATUS_CONN_AP_NOT_FOUND,
+
+       /** Last connection status. Please place all the status codes before this one. */
+       WIFI_STATUS_CONN_LAST_STATUS,
+       WIFI_STATUS_DISCONN_FIRST_STATUS = WIFI_STATUS_CONN_LAST_STATUS
 };
 
 /** Wi-Fi disconnect reason codes. To be overlaid on top of \ref wifi_status
@@ -384,7 +388,7 @@ enum wifi_conn_status {
  */
 enum wifi_disconn_reason {
        /** Unspecified reason */
-       WIFI_REASON_DISCONN_UNSPECIFIED = 0,
+       WIFI_REASON_DISCONN_UNSPECIFIED = WIFI_STATUS_DISCONN_FIRST_STATUS,
        /** Disconnected due to user request */
        WIFI_REASON_DISCONN_USER_REQUEST,
        /** Disconnected due to AP leaving */
```
WDYT?

Reply:
When I initially I added I kind of mixed protocol + usability, don't want to have 121 mapping with protocol as they aren't user friendly, but we can take one step ahead and have a generic status/reason so that both can be applicable to connection and disconnection.

Reply:
Ok, I will check if we can have generic status which can be applicable to both.